pyanaconda/ui/tui/spokes/storage.py

```diff
@@ -576,3 +596,256 @@ def input(self, args, key):
+        self._container = None
+        self._md = mount_data
+        self.title = N_("Configure device: %s") % mount_data.device
+
+    @property
+    def indirect(self):
+        return True
+
+    def refresh(self, args=None):
+        NormalTUISpoke.refresh(self, args)
+
+        self._container = ListColumnContainer(1)
+
+        fmt = get_format(self._md.format)
+        if fmt and fmt.mountable:
+            self._container.add(TextWidget("Mount point: %s" % self._md.mount_point or "none"), self._assign_mount_point, self._md)
+        else:
+            self._container.add(TextWidget("Mount point: [%s]" % fmt.name), lambda x: self.redraw(), None)
+        self._container.add(TextWidget("Format: %s" % self._md.format or "none"), self._set_format, self._md)
+        if ((self._md.orig_format and self._md.orig_format != self._md.format) or self._md.mount_point == "/"):
+            # changing format implies reformat and so does "/" mount point
+            self._container.add(CheckboxWidget(title="Reformat", completed=self._md.reformat), lambda x: self.redraw(), None)
+        else:
+            self._container.add(CheckboxWidget(title="Reformat", completed=self._md.reformat), self._switch_reformat, self._md)
+
+        self.window.add_with_separator(self._container)
+
+        message = _("Choose from above to assign mount point and/or set format.")
+        self.window.add_with_separator(TextWidget(message))
+
+    def input(self, args, key):
+        """ Grab the choice and update things. """
+        if not self._container.process_user_input(key):
+            return super(ConfigureDeviceSpoke, self).input(args, key)
+
+        return InputState.PROCESSED
+
+    def apply(self):
+        # nothing to do here, the callbacks below directly modify the data
+        pass
+
+    def _switch_reformat(self, md):
+        md.modified = True
+        md.reformat = not md.reformat
+        self.redraw()
+
+    def _set_format(self, md):
+        md.modified = True
+        dialog = SetFormatDialog(self.data, self.storage, self.payload, self.instclass, md)
+        ScreenHandler.push_screen(dialog)
+
+    def _assign_mount_point(self, md):
+        md.modified = True
+        dialog = SetMountPointDialog(self.data, self.storage, self.payload, self.instclass, md)
+        ScreenHandler.push_screen(dialog)
+
+class SetFormatDialog(EditTUIDialog):
+    """ Set format for a device """
+    category = SystemCategory
+
+    def __init__(self, data, storage, payload, instclass, mount_data):
+        EditTUIDialog.__init__(self, data, storage, payload, instclass)
+        self.title = N_("Configure device format")
+        self._md = mount_data
+
+    @property
+    def indirect(self):
+        return True
+
+    def prompt(self, args=None):
+        if self.value is None:  # first run or nothing entered
+            return Prompt(_("New format for the device%s") % (" [%s]" % self._md.format if self._md.format else ""))
```

Comment:
First of all it is not clear what you are doing when you have `%` within `%`.
Second I would move the space to the original string so that `...device%s` will be `...device %s`.

Reply:
But that would mean there would always be a space after ``device`` which is exactly the result I want to prevent.